Sources/Workspace/Workspace.swift

```diff
@@ -728,16 +749,8 @@ public class Workspace {
             updateConstraints += pinsStore.createConstraints()
         }
 
-        // Add unversioned constraint for edited packages.
-        for (externalManifest, managedDependency) in currentManifests.dependencies {
-            switch managedDependency.state {
-            case .checkout: continue
-            case .unmanaged, .edited: break
-            }
-            let specifier = RepositorySpecifier(url: externalManifest.url)
-            let dependencies = externalManifest.package.dependencyConstraints()
-            updateConstraints += [RepositoryPackageConstraint(container: specifier, requirement: .unversioned(dependencies))]
-        }
+        // Add unversioned constraints for edited packages.
+        updateConstraints += currentManifests.unversionedConstraints()
```

Comment:
I'm at a loss on understanding this comment I'm afraid. Which sets of dependencies are being duplicated?